app/src/main/java/io/apicurio/registry/store/RegistryStorage.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.store;
+
+import java.util.List;
+
+import io.apicurio.registry.rest.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.beans.EditableMetaData;
+import io.apicurio.registry.rest.beans.Rule;
+import io.apicurio.registry.rest.beans.VersionMetaData;
```

Comment:
I know these were generated, but I have a few questions:
Why are there two MetaData beans?

Should the possible metadata values be hardcoded? Can user set their own metadata value? I would rather see the persistence layer does not assume a fixed list of possible metadata values, and would use a key-value approach, the same for the rules (and rule configuration).
This doesn't have to mean that the REST beans have to change, rather the persistence layer should not accept/return REST beans directly, but work with a more general representation where the metadata keys & values are not fixed.

Reply:
I think you're probably right - the storage layer probably shouldn't deal in REST beans.  I will change the storage interface to have its own DTOs where necessary and just use standard structures (like `Map`) where possible.